Building with custom cmake -DCMAKE_INSTALL_PREFIX= fails on ubuntu 16.10

I've used `apt` to install awesome 3.5.9. I'd like to be able to run 4.0 side-by-side in order to verify that I won't run into problems with my setup. I was expecting that doing `cmake -DCMAKE_INSTALL_PREFIX=/opt/awesome`  and then `make` followed by `make package` would pop out a nice `.deb` for me to test with. However, I get the following error during `make`

```
-- Build files have been written to: /home/emikkva/src/awesome-4.0
[  1%] Built target generated_sources
[  2%] Generating awesomerc.lua, docs/05-awesomerc.md
lua: /home/emikkva/src/awesome-4.0/docs/05-awesomerc.md.lua:247: attempt to index local 'rc' (a nil value)
stack traceback:
	/home/emikkva/src/awesome-4.0/docs/05-awesomerc.md.lua:247: in main chunk
	[C]: in ?
CMakeFiles/generate_awesomerc.dir/build.make:60: recipe for target 'awesomerc.lua' failed
make[2]: *** [awesomerc.lua] Error 1
CMakeFiles/Makefile2:169: recipe for target 'CMakeFiles/generate_awesomerc.dir/all' failed
make[1]: *** [CMakeFiles/generate_awesomerc.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```
I'm not entirely sure how I should be using the `cmake` environment variables here. It seems that if I use the `CMAKE_INSTALL_PREFIX`, something during the subsequent `make` removes the `awesomerc.lua` from the build directory before the step above is executed (and fails). If I run `make` in a freshly extracted folder, `awesomerc.lua` is not removed and the build process succeeds.

EDIT:
The idea of using `CMAKE_INSTALL_PREFIX` comes from #1048 and may be completely the wrong way of achieving what I want to do – customize the installation location.

--- a/docs/01-readme.md
+++ b/docs/01-readme.md
@@ -19,10 +19,16 @@ After building is finished, you can either install via `make install`:
 or by auto-generating a .deb or .rpm package, for easy removal later on:
 
     make package
-    
+
     sudo dpkg -i awesome-x.y.z.deb
     # or
     sudo rpm -Uvh awesome-x.y.z.rpm
+
+NOTE: awesome uses [`cmake`](https://cmake.org) to build. In case you want to
+pass arguments to cmake, please use the `CMAKE_ARGS` environment variable. For
+instance:
+
+    CMAKE_ARGS="-DCMAKE_INSTALL_PREFIX=/opt/awesome" make
 
 ### Build dependencies
 

